api/tests/routes/backoffice/accounts_test.py

```diff
@@ -2403,3 +2413,32 @@ def test_get_tunnel(self):
         users_factories.DepositGrantFactory(user=user)
         tunnel_end = _get_tunnel(user, eligibility_history, fraud_reviews_desc)
         assert tunnel_end["progress"] == 100
+
+
+class AnonymizePublicAccountTest(PostEndpointHelper):
+    endpoint = "backoffice_web.public_accounts.anonymize_public_account"
+    endpoint_kwargs = {"user_id": 1}
+    needed_permission = perm_models.Permissions.ANONYMIZE_PUBLIC_ACCOUNT
+
+    @patch("pcapi.routes.backoffice.accounts.blueprint.users_api.anonymize_user")
+    def test_anonymize_public_account(
+        self,
+        mock_anonymize_user,
+        legit_user,
+        authenticated_client,
+    ):
+        user = users_factories.UserFactory()
+
+        history_factories.ActionHistoryFactory(
+            actionType=history_models.ActionType.USER_ANONYMIZED,
+            authorUser=legit_user,
+            userId=user.id,
+        )
+
+        response = self.post_to_endpoint(authenticated_client, user_id=user.id)
+        assert response.status_code == 302
+
+        expected_url = url_for("backoffice_web.public_accounts.get_public_account", user_id=user.id, _external=True)
+        assert response.location == expected_url
+
+        mock_anonymize_user.assert_called_once()
```

Comment:
Mieux : vérifier que c'est appelé avec les bons paramètres
```suggestion
        mock_anonymize_user.assert_called_once_with(user, legit_user, True)
```
À partir du moment où tu utilises le `mock`, vérifie que l'auteur que tu passes en paramètre est bien testé dans un test unitaire de la fonction `anonymize_user`.

Reply:
account_test ligne 2444
`        mock_anonymize_user.assert_called_once_with(user, legit_user, True)`

FAIT, interressant ainsi on peut repprendre les attributs de la fonction original,

Reply:
ne fonctionne pas , apres avoir vue avec Patrick, test sans mock a faire n marqué non résolue

Reply:
```
    def test_anonymize_public_account(
        self,
        legit_user,
        authenticated_client,
    ):
        user = users_factories.UserFactory()

        response = self.post_to_endpoint(authenticated_client, user_id=user.id)
        assert response.status_code == 302

        expected_url = url_for("backoffice_web.public_accounts.get_public_account", user_id=user.id, _external=True)
        assert response.location == expected_url

        assert "Anonymous" in user.firstName
        assert user.roles == [users_models.UserRole.ANONYMIZED]
        history = history_models.ActionHistory.query.one_or_none()
        assert history.actionType == history_models.ActionType.USER_ANONYMIZED
```

FAIT